src/Model/Scope/BasicCondition.php

```diff
@@ -0,0 +1,340 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model\Scope;
+
+use atk4\core\ReadableCaptionTrait;
+use atk4\data\Exception;
+use atk4\data\Field;
+use atk4\dsql\Expression;
+use atk4\dsql\Expressionable;
+
+class BasicCondition extends AbstractCondition
+{
+    use ReadableCaptionTrait;
+
+    /**
+     * Stores the condition key.
+     *
+     * @var string|Field|Expression
+     */
+    public $key;
+
+    /**
+     * Stores the condition operator.
+     *
+     * @var string
+     */
+    public $operator;
+
+    /**
+     * Stores the condition value.
+     */
+    public $value;
+
+    protected static $opposites = [
+        '=' => '!=',
+        '!=' => '=',
+        '<' => '>=',
+        '>' => '<=',
+        '>=' => '<',
+        '<=' => '>',
+        'LIKE' => 'NOT LIKE',
+        'NOT LIKE' => 'LIKE',
+        'IN' => 'NOT IN',
+        'NOT IN' => 'IN',
+        'REGEXP' => 'NOT REGEXP',
+        'NOT REGEXP' => 'REGEXP',
+    ];
+
+    protected static $dictionary = [
+        '=' => 'is equal to',
+        '!=' => 'is not equal to',
+        '<' => 'is smaller than',
+        '>' => 'is greater than',
+        '>=' => 'is greater or equal to',
+        '<=' => 'is smaller or equal to',
+        'LIKE' => 'is like',
+        'NOT LIKE' => 'is not like',
+        'IN' => 'is one of',
+        'NOT IN' => 'is not one of',
+        'REGEXP' => 'is regular expression',
+        'NOT REGEXP' => 'is not regular expression',
+    ];
+
+    protected static $skipValueTypecast = [
+        'LIKE',
+        'NOT LIKE',
+        'REGEXP',
+        'NOT REGEXP',
+    ];
+
+    public function __construct($key, $operator = null, $value = null)
+    {
+        if (func_num_args() == 1 && is_bool($key)) {
+            if ($key) {
+                return;
+            }
+
+            $key = new Expression('false');
+        }
+
+        if (func_num_args() == 2) {
+            $value = $operator;
+            $operator = '=';
+        }
+
+        $this->key = $key;
+        $this->operator = $operator;
+        $this->value = $value;
+    }
+
+    public function onChangeModel(): void
+    {
+        if ($model = $this->getModel()) {
+            // if we have a definitive scalar value for a field
+            // sets it as default value for field and locks it
+            // new records will automatically get this value assigned for the field
+            // @todo: consider this when condition is part of OR scope
+            if ($this->operator === '=' && !is_object($this->value) && !is_array($this->value)) {
```

Comment:
this is interesting function, however Scopes should not change a default value, at least there should be a switch in Model/Field/Persistence to enable/disable this.

I think this has the following issues now:
- his can be also a security issue, it may set some secret value we expect (even password/hash :D )
- what if another scope is attached later?
- system property may hide column from Table, etc. which may not be desired (even very confusing - if I filter with `=` on table, should it hide that column? No!)

This should be probably completely removed and implemented in UI Persistence only.

Reply:
This is an existing feature in ATK, I have not introduced it.
We can consider updating this functionality.

Reply:
to be solved later, issue https://github.com/atk4/data/issues/662